libbeat/cmd/instance/ilm.go

```diff
@@ -0,0 +1,168 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package instance
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+type ilmConfig struct {
+	WriteAlias string `config:"write_alias" `
+	Name       string `config:"index.name"`
+	Pattern    string `config:"index.pattern"`
+}
+
+// Build and return a callback to load index template into ES
+func (b *Beat) writeAliasLoadingCallback() (func(esClient *elasticsearch.Client) error, error) {
```

Comment:
With the command here I create the alias and the index at the same time. So you would split it up into 2 calls? Wouldn't that mean in case of such a race I would still have an exception when creating the index?

Reply:
Good point! Then the best bet might be to just ignore the _alias_already_exists_ error returned from ES, and only check for other errors.

Reply:
The way the implementation looks now is that it first checks if it exists and if not creates it. It could still happen that several Beats try to create it at the same time but I would prefer to keep the code simple here.

Reply:
So with this implementation you basically risk that a beat cannot startup because of a race condition with another beat. Wouldn't it be better then to at least catch a `400` status code and ignore it https://github.com/elastic/beats/pull/7963/files#diff-2da066f8f6a753557247f81119034e25R95?

Reply:
+1 for ignoring (but logging) alias_already_exists errors.